crates/cairo-lint-core/src/plugin.rs

```diff
@@ -20,17 +22,55 @@ pub struct CairoLint;
 pub enum CairoLintKind {
     DestructMatch,
     MatchForEquality,
+    DoubleComparison,
     Unknown,
 }
 
 pub fn diagnostic_kind_from_message(message: &str) -> CairoLintKind {
     match message {
-        single_match::DESTRUCT_MATCH => CairoLintKind::DestructMatch,
-        single_match::MATCH_FOR_EQUALITY => CairoLintKind::MatchForEquality,
+        CairoLint::DESTRUCT_MATCH => CairoLintKind::DestructMatch,
+        CairoLint::MATCH_FOR_EQUALITY => CairoLintKind::MatchForEquality,
+        CairoLint::DOUBLE_COMPARISON => CairoLintKind::DoubleComparison,
         _ => CairoLintKind::Unknown,
     }
 }
 
+impl CairoLint {
+    const DESTRUCT_MATCH: &'static str =
+        "you seem to be trying to use `match` for destructuring a single pattern. Consider using `if let`";
+    const MATCH_FOR_EQUALITY: &'static str =
+        "you seem to be trying to use `match` for an equality check. Consider using `if`";
+    const DOUBLE_COMPARISON: &'static str =
+        "redundant double comparison found. Consider simplifying to a single comparison.";
+
+    pub fn check_double_comparison(&self, db: &dyn SyntaxGroup, expr: &Expr, diagnostics: &mut Vec<PluginDiagnostic>) {
+        if let Expr::Binary(binary_op) = expr {
+            let lhs = binary_op.lhs(db);
+            let rhs = binary_op.rhs(db);
+
+            if let (Some(lhs_op), Some(rhs_op)) =
+                (self.extract_binary_operator(&lhs, db), self.extract_binary_operator(&rhs, db))
+            {
+                if (matches!(lhs_op, BinaryOperator::EqEq(_)) && matches!(rhs_op, BinaryOperator::LT(_)))
+                    || (matches!(lhs_op, BinaryOperator::LT(_)) && matches!(rhs_op, BinaryOperator::EqEq(_)))
+                    || (matches!(lhs_op, BinaryOperator::GE(_)) && matches!(rhs_op, BinaryOperator::GT(_)))
+                    || (matches!(lhs_op, BinaryOperator::LE(_)) && matches!(rhs_op, BinaryOperator::LT(_)))
```

Comment:
check [here](https://github.com/rust-lang/rust-clippy/blob/5da97d006e33a4458f467e41ca43085f5e5b20ba/clippy_lints/src/operators/double_comparison.rs#L38-L50) to get all the patterns